aws/resource_aws_cloudfront_key_group_test.go

```diff
@@ -0,0 +1,282 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSCloudFrontKeyGroupConfigItems(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudFrontKeyGroupExistence(resourceName),
+					resource.TestCheckResourceAttr("aws_cloudfront_key_group.test", "items.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudFrontKeyGroupExistence(r string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[r]
+		if !ok {
+			return fmt.Errorf("not found: %s", r)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no Id is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).cloudfrontconn
+
+		input := &cloudfront.GetKeyGroupInput{
+			Id: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetKeyGroup(input)
+		if err != nil {
+			return fmt.Errorf("error retrieving CloudFront key group: %s", err)
+		}
+		return nil
+	}
+}
+
+func testAccCheckCloudFrontKeyGroupDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).cloudfrontconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_cloudfront_key_group" {
+			continue
+		}
+
+		input := &cloudfront.GetKeyGroupInput{
+			Id: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetKeyGroup(input)
+		if isAWSErr(err, cloudfront.ErrCodeNoSuchResource, "") {
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("CloudFront key group (%s) was not deleted", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccAWSCloudFrontKeyGroupConfigBase(rInt int) string {
+	return fmt.Sprintf(`
+resource "aws_cloudfront_public_key" "test" {
+  comment     = "test key"
+  encoded_key = file("test-fixtures/cloudfront-public-key.pem")
+  name        = "tf-acc-test-%d"
+}
+`, rInt)
+}
+
+func testAccAWSCloudFrontKeyGroupConfig(rInt int) string {
+	return testAccAWSCloudFrontKeyGroupConfigBase(rInt) + fmt.Sprintf(`
+resource "aws_cloudfront_key_group" "test" {
+  comment = "test key group"
+  items   = [aws_cloudfront_public_key.test.id]
+  name    = "tf-acc-test-%d"
+}
+`, rInt)
+}
+
+func testAccAWSCloudFrontKeyGroupConfigComment(rInt int, comment string) string {
+	return testAccAWSCloudFrontKeyGroupConfigBase(rInt) + fmt.Sprintf(`
+resource "aws_cloudfront_key_group" "test" {
+  comment = %q
+  items   = [aws_cloudfront_public_key.test.id]
+  name    = "tf-acc-test-%d"
+}
+`, comment, rInt)
+}
+
+func testAccAWSCloudFrontKeyGroupConfigItems(rInt int) string {
+	return testAccAWSCloudFrontKeyGroupConfigBase(rInt) + fmt.Sprintf(`
+resource "aws_cloudfront_public_key" "test2" {
+  comment     = "second test key"
+  encoded_key = file("test-fixtures/cloudfront-public-key.pem")
+  name        = "tf-acc-test-second-%d"
+}
+
+resource "aws_cloudfront_key_group" "test" {
+  comment = "test key group"
+  items   = [aws_cloudfront_public_key.test.id, aws_cloudfront_public_key.test2.id]
+  name    = "tf-acc-test-%d"
+}
+`, rInt, rInt)
```

Comment:
Go's `fmt` package supports indexed verbs, such as `%[1]d` that allow you not repeat parameters.